core/src/impl/Kokkos_ViewCtor.hpp

```diff
@@ -55,6 +55,25 @@ struct WithoutInitializing_t {};
 struct AllowPadding_t {};
 struct NullSpace_t {};
 
+template <typename>
+struct is_view_ctor_property : public std::false_type {};
+
+template <>
+struct is_view_ctor_property<WithoutInitializing_t> : public std::true_type {};
+
+template <>
+struct is_view_ctor_property<AllowPadding_t> : public std::true_type {};
+
+template <>
+struct is_view_ctor_property<NullSpace_t> : public std::true_type {};
+
+template <typename T>
+struct is_without_initializing {
+  enum : bool {
+    value = std::is_same<T, Kokkos::Impl::WithoutInitializing_t>::value
+  };
+};
```

Comment:
```suggestion
template <typename T>
struct is_without_initializing
    : std::integral_constant<
          bool, std::is_same<T, Kokkos::Impl::WithoutInitializing_t>::value> {};
```

Reply:
Also you should define these in `Impl::`

Reply:
And now that I see how it is used I'd actually rather have you not introduce it.

Reply:
What are you suggesting to do for detecting this attribute instead?

Reply:
Good point.  You'd need something like `std::disjunction` or a fold expression, both of which require C++17.

Reply:
Also, now that I see that again
```C++
template <typename T>
struct is_without_initializing : typename std::is_same<T, Kokkos::Impl::WithoutInitializing_t>::type {};